hi-ml-histopathology/testhisto/testhisto/mocks/tiles_datamodule.py

```diff
@@ -0,0 +1,165 @@
+
+    N_CLASSES = MockTilesDataset.N_CLASSES
+
+    METADATA_POSSIBLE_VALUES: dict = {
+        DATA_PROVIDER: ["site_0", "site_1"],
+        ISUP_GRADE: {
+            0: ["0+0", "negative"],
+            4: ["4+4", "5+3", "3+5"],
+            1: ["3+3"],
+            3: ["4+3"],
+            2: ["3+4"],
+            5: ["4+5", "5+4", "5+5"],
+        },
+    }
+
+    METADATA_COLUMNS = tuple(METADATA_POSSIBLE_VALUES.keys())
+
+    def __init__(self, img_size: int = 224, **kwargs: Any) -> None:
+        self.img_size = img_size
+        super().__init__(**kwargs)
+        assert (
+            self.n_slides >= self.N_CLASSES
+        ), f"The number of slides should be >= self.N_CLASSES (i.e., {self.N_CLASSES})"
+        assert (self.img_size // self.tile_size) ** 2 >= self.n_tiles, (
+            f"The image of size {self.img_size} can't contain more than {(self.img_size // self.tile_size)**2} tiles."
+            "Choose a number of tiles n_tiles <= (img_size // tile_size)**2 "
+        )
+
+    def create_mock_metadata_dataframe(self) -> pd.DataFrame:
+        """Create a mock dataframe with random metadata."""
+        df = pd.DataFrame(
+            columns=[
+                self.SLIDE_ID_COLUMN,
+                self.TILE_ID_COLUMN,
+                self.IMAGE_COLUMN,
+                self.MASK_COLUMN,
+                self.TILE_X_COLUMN,
+                self.TILE_Y_COLUMN,
+                self.OCCUPANCY,
+                self.DATA_PROVIDER,
+                self.ISUP_GRADE,
+                self.GLEASON_SCORE,
+            ]
+        )
+        grid = [
+            f"{i}_{j}" for i in range(self.img_size // self.tile_size) for j in range(self.img_size // self.tile_size)
+        ]
+        coords = list(
+            map(lambda x: list(map(int, x.split("_"))), np.random.choice(grid, size=self.n_tiles, replace=False))
+        )
+
+        isup_grades = np.tile(
+            list(self.METADATA_POSSIBLE_VALUES[self.ISUP_GRADE].keys()), self.n_slides // self.N_CLASSES + 1
+        )
+
+        for slide_id in range(self.n_slides):
+            data_provider = np.random.choice(self.METADATA_POSSIBLE_VALUES[self.DATA_PROVIDER])
+            isup_grade = isup_grades[slide_id]
+            gleason_score = np.random.choice(self.METADATA_POSSIBLE_VALUES[self.ISUP_GRADE][isup_grade])
+            for tile_id in range(self.n_tiles):
+                tile_x = coords[tile_id][0] * self.tile_size
+                tile_y = coords[tile_id][1] * self.tile_size
+                df.loc[(slide_id + 1) * tile_id] = [
+                    f"_{slide_id}",
+                    f"_{slide_id}.{tile_x}x_{tile_y}y",
+                    f"_{slide_id}/train_images/{tile_x}x_{tile_y}y.png",
+                    f"_{slide_id}/train_images/{tile_x}x_{tile_y}y_mask.png",
+                    tile_x,
+                    tile_y,
+                    1.0,
+                    data_provider,
+                    isup_grade,
+                    gleason_score,
```

Comment:
Minor: building a dictionary using as keys the column names defined in rows 111-121 would have been a bit more readable and less prone to mismatch in the fields order, a df can be created from the dictionary. just an idea, no need to change now.